index.bs

```diff
@@ -170,19 +182,68 @@ When an image item is of type <dfn value export for="AV1 Image Item Type">av01</
+
+If an [=AV1 Multi-Layered Image Item=] is not associated with an 'lsel' property, the 'ispe' property shall describe the dimension of the output image of the operating point whose index is 0. Renderers are free to render either: only the output image of the [=Operating Point=] matching best its decoding capabilities, or to render all output images of all the [=Operating Points=] processed, resulting in a form of progressive decoding.
+
+NOTE: The dimensions of the final or intermediate output images might not match the ones given in the 'ispe' property. Renderers are expected to scale the output image to match the 'ispe' property.
+
+NOTE: When such a progressive decoding is not desired (e.g. for multi-view images) or when a controlled rendering of the different output images is desired (e.g. grid of multi-layered images), multiple [=AV1 Image Items=] sharing the same [=AV1 Image Item Data=] can be created and associated with different 'lsel' properties, each with a different value of 'layer_id'. In the case of an image grid composed of 4 layered images, each made of 2 operating points, controlled rendering could be achieved by creating a set of items as follows: 4 items each corresponding to the low quality/resolution image of the first operating point of the 4 multi-layered images, i.e. with 'layer_id' set to 0; 4 items corresponding the second operating point, i.e. with 'layer_id' set to 1; a grid item composing the first set of 4 images; another grid item composing the second set of 4 images.
+
+<h5 id ="layered-image-selector-property">Layered Image Indexing Property</h5>
+
+<h6 id ="layered-image-selector-property-definition" class="no-toc">Definition</h6>
+
+<pre class="def">
+  Box Type:       <dfn value export for="AV1LayeredImageIndexingProperty">a1lx</dfn>
+  Property type:  Descriptive item property
+  Container:      <code>ItemPropertyContainerBox</code>
+  Mandatory:      No
+  Quantity:       One
+</pre>
+
+<h6 id ="layered-image-selector-property-description" class="no-toc">Description</h6>
+
+The <dfn>AV1LayeredImageIndexingProperty</dfn> provides the size in bytes of each layer in the image data. It enables determining the byte ranges required to process an [=Operating Point=].
+
+The [=AV1LayeredImageIndexingProperty=] property may be associated to an [=AV1 Multi-Layered Image Item=]. It shall not be marked as essential. It shall not be associated with [=AV1 Image Items=] that are not using multiple layers.
+
+<h6 id ="layered-image-selector-property-syntax" class="no-toc">Syntax</h6>
+
+```
+class AV1LayeredImageIndexingProperty extends ItemFullProperty('a1lx', version = 0, flags) {
+    FieldLength = ((flags & 1) + 1) * 16;
+    unsigned int(FieldLength)[4] layer_size;
```

Comment:
I guess the offset is implicitly the sum of all previous layer sizes?

Reply:
This is explicitly tailored for AV1. Do we want to make it slightly more generic so that it can also work for AV2?

Do we want the array to be variable size? Doing so would give two things:
1. You would only have to specify the layers that actually exist.
2. It would be future proof with AV2 if AV2 adds more layers than 4.

Reply:
It's an option. We can always define a new version of the box for AV2 as we de don't know how AV2 will be.

Reply:
Adding an array size field will make it clear which array entries are valid, without referring to the OperatingPointIdc bitmask. I guess it will also save a few bytes in the common cases.

Reply:
Cyril: I just realized that AV1 does not seem to require that a bitstream use contiguous spatial_id values. So it may be valid for a bitstream with two spatial layers to use spatial_id values 0 and 2. So I think your current design is good. Unused spatial_id values can be indicated by 0 in the layer_size array.

Reply:
> Cyril: I just realized that AV1 does not seem to require that a bitstream use contiguous spatial_id values. So it may be valid for a bitstream with two spatial layers to use spatial_id values 0 and 2. So I think your current design is good. Unused spatial_id values can be indicated by 0 in the layer_size array.

👍 Sounds like a strong justification for the current design.